### PR TITLE
properly detect errors in wrapping/unwrapping window.addEventListener

### PR DESCRIPTION
**Pull request by davidsbailey**

### CODE REVIEW

File: apps/test/util/wrappedEventListener.js [hunk 1]
```diff
@@ -10,15 +7,22 @@ module.exports = {
    * Attach our wrapper, tracking all added listeners
    */
   attach: function () {
-    if (original) {
+    if (window.addEventListener.toString().indexOf('[native code]') === -1) {
```
Comment (Hamms) [inline apps/test/util/wrappedEventListener.js]: Might be worth pulling this out into an "isNativeCode" helper, just to make it clearer what's going on here.

File: apps/test/util/wrappedEventListener.js [hunk 2]
```diff
@@ -26,11 +30,15 @@ module.exports = {
...
   detach: function () {
-    window.addEventListener = original;
-    original = null;
-    listeners.forEach(function (argList) {
+    if (!window.originalAddEventListener ||
+        window.originalAddEventListener.toString().indexOf('[native code]') === -1) {
+      throw new Error('window.originalAddEventListener is not a native function');
+    }
+    window.addEventListener = window.originalAddEventListener;
+    window.originalAddEventListener = null;
```
Comment (Hamms) [inline apps/test/util/wrappedEventListener.js:41]: superminor nit: `window.originalAddEventListener = undefined;` will return us to a state slightly closer to default

**Comment by Hamms** [comment]
Were you able to consistently repro the error this is fixing?

**Comment by Hamms** [comment]
LGTM!

**Comment by davidsbailey** [comment]
For the record I was not able to repro the error that this is addressing, namely the following bottomless stack trace:

```
[TRUNCATED]il/wrappedEventListener.js:21
      at test/util/wrappedEventListener.js:21
      at test/util/wrappedEventListener.js:21
```

**Pull request merged**